Studies/source/EfficiencyStudy.cxx

```diff
@@ -0,0 +1,193 @@
+        canvas.Clear();
+    }
+
+    void CreateEfficiencies(const TH1& selection_hist, const std::string& channel,
+                            const std::string& hist_name, int shift, bool create_relative, double sf)
+    {
+        int N = selection_hist.GetNbinsX()-shift;
+        TH1D deno("", "", N, 0, N);
+        TH1D nume(deno), total(deno), empty(deno);
+
+        for(int n = 1; n <= N; ++n){
+            double totalWithWeight = std::ceil(selection_hist.GetBinContent(1)*sf);
+
+            deno.SetBinContent(n, selection_hist.GetBinContent(n));
+            deno.SetBinError(n, selection_hist.GetBinError(n));
+            nume.SetBinContent(n, selection_hist.GetBinContent(n+shift));
+            nume.SetBinError(n, selection_hist.GetBinError(n+shift));
+            total.SetBinContent(n, totalWithWeight);
+            total.SetBinError(n, std::sqrt(totalWithWeight));
+            nume.GetXaxis()->SetBinLabel(n, selection_hist.GetXaxis()->GetBinLabel(n+shift));
+            empty.GetXaxis()->SetBinLabel(n, selection_hist.GetXaxis()->GetBinLabel(n+shift));
+
+        }
+        CreateEfficiency(nume, total, "eff_abs", channel, hist_name);
+        if(create_relative)
+        CreateEfficiency(nume, deno, "eff_rel", channel, hist_name);
+       }
+
+     void Run()
+     {
+         auto file = root_ext::OpenRootFile(args.input_file());
+
+         std::vector<ChannelDesc> Channels = {
+             ChannelDesc("eTau", {"events","SignalElectrons_Central", "SignalTaus_Central", "jets_Central" }, 0.23),
+             ChannelDesc("muTau", {"events","SignalMuons_Central", "SignalTaus_Central", "jets_Central"}, 0.23),
+             ChannelDesc("tauTau", {"events","SignalTaus_Central", "jets_Central"}, 0.54),
+         };
+
+         //Creating the efficiency histogrmas after events
+
+        for(const auto& entry : Channels) {
+            const std::string& channel_name = entry.channel;
+            const std::vector<std::string> hist_name = entry.histograms;
+            const double divide_by_br = entry.branching_ratio;
+
+            for(unsigned long i = 1; i < hist_name.size(); i++){
+
+                const std::string full_hist_name = channel_name + "_stat/Selection_" + hist_name.at(i);
+                auto selection_hist = std::shared_ptr<TH1D>(root_ext::ReadObject<TH1D>(*file, full_hist_name));
+                    
+                CreateEfficiencies(*selection_hist, channel_name, hist_name.at(i), 1, true, 1);
+
+            }
+
+            //Creating the efficiency histogrmas for events
+
+            const std::string Events = channel_name + "_stat/Selection_" + hist_name.at(0);
+            auto selection_events = std::shared_ptr<TH1D>(root_ext::ReadObject<TH1D>(*file, Events));
+
+            CreateEfficiencies(*selection_events, channel_name, hist_name.at(0), 1, true, 1);
+            CreateEfficiencies(*selection_events, channel_name, hist_name.at(0) + "BR" , 2, false, divide_by_br);
```

Comment:
Don't create separate code for the first histo: store boolean indicators in the ChannelDesc structure.